frontend/src/views/Profile.vue

```diff
@@ -1,15 +1,303 @@
+        <div class="lines"></div>
+      </div>
+      <div>
+        <div v-if="!displayGraph">No data yet!</div>
+        <div v-else>
+          <div id="graph">
+            <VueBarGraph :barColor="'#3aeddf'" :customLabels="['This week','Last week','2 Weeks Ago','3 Weeks Ago']"
+                         :height="200"
+                         :points="this.leaderboardInfo"
+                         :showValues="true"
+                         :showXAxis="true"
+                         :showYAxis="true"
+                         :textAltColor="'white'"
+                         :textColor="'white'"
+                         :use-custom-labels="true"
+                         :width="widthCalc()"
+            />
+          </div>
+        </div>
+      </div>
+    </div>
+  </div>
 </template>
 
 <script>
-export default {
+import axios from "axios";
+import VueBarGraph from "vue-bar-graph";
+import CommunitiesList from "../components/communities/CommunitiesList.vue";
 
-}
+export default {
+  components: {
+    VueBarGraph,
+    CommunitiesList,
+  },
+  mounted() {
+    this.getUserDetails();
+  },
+  methods: {
+    widthCalc() {
+      return window.innerWidth - 100;
+    },
+    getUserDetails() {
+      axios
+          .get("/api/users/me")
+          .then((response) => {
+            this.firstName = response.data.first_name;
+            this.secondName = response.data.last_name;
+            this.username = response.data.username;
+            this.isTeacher = response.data.is_teacher;
+            this.leaderboardPosition = response.data.leaderboard_position;
+            this.points = response.data.points;
+            var recents = response.data.graphs.recent;
+            this.leaderboardInfo = [
+              recents[0].points,
+              recents[1].points,
+              recents[2].points,
+              recents[3].points,
+            ];
+            if (
+                JSON.stringify(this.leaderboardInfo) == JSON.stringify([0, 0, 0, 0])
+            ) {
+              this.displayGraph = false;
+            }
+            for (
+                var i = 0;
+                i < response.data.graphs.top_communities.length;
+                i++
+            ) {
+              this.communities.push(
+                  response.data.graphs.top_communities[i].community
+              );
+            }
+            this.initials =
+                this.firstName.substring(0, 1) + this.secondName.substring(0, 1);
+            this.bio = response.data.descrtiption;
+          })
+        .catch((error) => {
+          console.log(error);
+        });
+    },
+  },
+  data() {
+    return {
+      id: 0,
+      username: "",
+      firstName: "",
+      secondName: "",
+      isTeacher: false,
+      points: 0,
+      initials: "",
+      leaderboardPosition: 0,
+      leaderboardInfo: [],
+      communities: [],
+      bio: "",
+      displayGraph: true,
+      labels: ["This week", "Last week", "2 Weeks Ago", "3 Weeks Ago"],
```

Comment:
You could compile all of these fields into one object. Then you dont have to use `this.first_name`, `this.last_name`, etc. You could just use `this.user = response.data.user`

Reply:
honestly these dont impact functionallity so this should get merged as we need this for other things and we can change them on another branch

Reply:
I am aware. Its just a NIT